### PR TITLE
[WIP] Android Subscription Proration

### PR DESCRIPTION
**Pull request by fierysolid**
Work in Progress. Found a bug with downgrades

### CODE REVIEW

**Comment by dooboolab** [comment]
@fierysolid Is this ready?

**Comment by fierysolid** [comment]
Not yet, still working out some kinks. Should have it ready EOD though.

**Comment by dooboolab** [comment]
@fierysolid NP you may slow it down. Tell me when it is ready, I'll review this.

File: android/src/main/java/com/dooboolab/RNIap/RNIapModule.java [hunk 1]
```diff
@@ -347,7 +347,7 @@ public void onPurchaseHistoryResponse(@BillingClient.BillingResponse int respons
   }
 
   @ReactMethod
-  public void buyItemByType(final String type, final String sku, final String oldSku, final Promise promise) {
+  public void buyItemByType(final String type, final String sku, final String oldSku, final Integer prorationMode, final Promise promise) {
     final Activity activity = getCurrentActivity();
```
Comment (dooboolab) [inline android/src/main/java/com/dooboolab/RNIap/RNIapModule.java]: @fierysolid You need to be aware that `public void buyItemByType(final String type, final String sku, final String oldSku, final Promise promise)` and `public void buyItemByType(final String type, final String sku, final String oldSku, final Integer prorationMode, final Promise promise)` are different method in java. You should make another method and leave `buyItemByType` as it was before. Java isn't working like javascript and it actually treats method with the different number of parameters differently.

**Pull request merged**